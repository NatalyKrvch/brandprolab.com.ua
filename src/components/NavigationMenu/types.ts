export type MenuItemType = {
  _key: string;
  _type: 'menuItem';
  label: string;
  link: string;
};

export interface NavigationMenuProps {
  menuItems: MenuItemType[];
  className?: string;
}
