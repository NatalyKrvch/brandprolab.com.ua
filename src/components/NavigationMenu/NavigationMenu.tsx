import { twMerge } from 'tailwind-merge';

import { normalizeSectionHref } from '@/utils';

import { InternalLink } from '../Links';
import type { NavigationMenuProps } from './types';

const NavigationMenu = ({ menuItems, className }: NavigationMenuProps) => {
  if (!menuItems || menuItems.length === 0) return null;

  return (
    <nav aria-label="Головне меню" className={twMerge(className)}>
      <ul className="flex flex-col gap-8 desktop:flex-row">
        {menuItems.map(item => (
          <li key={item._key}>
            <InternalLink
              href={normalizeSectionHref(item.link)}
              prefetch={false}
              className={twMerge(
                'text-xl font-medium leading-none tracking-normal transition-colors',
                'text-white hover:text-teal-dark tablet:text-32 desktop:text-gray-dark',
                'rounded-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-teal-dark focus-visible:ring-offset-2 focus-visible:ring-offset-transparent',
              )}
            >
              {item.label}
            </InternalLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavigationMenu;
