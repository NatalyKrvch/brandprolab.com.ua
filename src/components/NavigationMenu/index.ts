export { default as NavigationMenu } from './NavigationMenu';
export * from './types';
