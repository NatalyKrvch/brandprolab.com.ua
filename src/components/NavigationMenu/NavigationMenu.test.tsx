import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import NavigationMenu from './NavigationMenu';
import type { MenuItemType } from './types';

jest.mock('../Links', () => ({
  InternalLink: ({ href, className, children, _prefetch, ...rest }: any) => (
    <a href={href} className={className} {...rest}>
      {children}
    </a>
  ),
}));

jest.mock('@/utils', () => ({
  normalizeSectionHref: (link: string) => {
    if (!link) return '#';
    const id = link.trim().replace(/^#/, '').replace(/\s+/g, '-');
    return `#${id}`;
  },
}));

const menuItemsMock: MenuItemType[] = [
  { _key: '1', _type: 'menuItem', label: 'Про нас', link: 'about' },
  { _key: '2', _type: 'menuItem', label: 'Послуги', link: 'services' },
  { _key: '3', _type: 'menuItem', label: 'Контакти', link: 'contacts' },
];

describe('NavigationMenu', () => {
  it('renders all menu items correctly', () => {
    render(<NavigationMenu menuItems={menuItemsMock} />);

    menuItemsMock.forEach(item => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
    });
  });

  it('applies custom className if provided', () => {
    render(
      <NavigationMenu menuItems={menuItemsMock} className="custom-class" />,
    );
    expect(screen.getByRole('navigation')).toHaveClass('custom-class');
  });

  it('normalizes hrefs by adding #', () => {
    render(<NavigationMenu menuItems={menuItemsMock} />);
    const links = screen.getAllByRole('link');
    expect(links.map(a => a.getAttribute('href'))).toEqual([
      '#about',
      '#services',
      '#contacts',
    ]);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<NavigationMenu menuItems={menuItemsMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
