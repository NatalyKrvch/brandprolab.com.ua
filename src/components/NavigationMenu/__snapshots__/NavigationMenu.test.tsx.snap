// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Головне меню"
    class=""
  >
    <ul
      class="flex flex-col gap-8 desktop:flex-row"
    >
      <li>
        <a
          class="text-xl font-medium leading-none tracking-normal transition-colors text-white hover:text-teal-dark tablet:text-32 desktop:text-gray-dark rounded-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-teal-dark focus-visible:ring-offset-2 focus-visible:ring-offset-transparent"
          href="#about"
        >
          Про нас
        </a>
      </li>
      <li>
        <a
          class="text-xl font-medium leading-none tracking-normal transition-colors text-white hover:text-teal-dark tablet:text-32 desktop:text-gray-dark rounded-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-teal-dark focus-visible:ring-offset-2 focus-visible:ring-offset-transparent"
          href="#services"
        >
          Послуги
        </a>
      </li>
      <li>
        <a
          class="text-xl font-medium leading-none tracking-normal transition-colors text-white hover:text-teal-dark tablet:text-32 desktop:text-gray-dark rounded-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-teal-dark focus-visible:ring-offset-2 focus-visible:ring-offset-transparent"
          href="#contacts"
        >
          Контакти
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
